source/fuzz/transformation_store.cpp

```diff
@@ -125,5 +126,18 @@ protobufs::Transformation TransformationStore::ToMessage() const {
   return result;
 }
 
+bool TransformationStore::StorageClassIsReadOnly(
+    SpvStorageClass storage_class) {
+  switch (storage_class) {
+    case SpvStorageClassInput:
+    case SpvStorageClassPushConstant:
+    case SpvStorageClassUniformConstant:
+      // Read-only: cannot store to it.
+      return true;
+    default:
+      return false;
+  }
+}
```

Comment:
There seems to already be `IsReadOnlyVariableShaders`. Could that be used? It looks more complex though.

Reply:
It appears that it only applies to OpVariable instructions.  I need to know whether a pointer is suitable for a store, and that pointer could have come from an access chain.

Looks like the code should be adaptable, though.  I have opened an issue (#3324) which I will reference from this code, and will refactor later if it turns out to be reusable.

Reply:
I have made a PR - #3325 - that should allow us to re-use this functionality.

Reply:
Adapted this PR to re-use that functionality.